File: pc-server/Models/FileItem.cs
namespace BluetoothFileServer.Models;

public class FileItem
{
    public string Name { get; set; } = "";
    public long Size { get; set; }
    public DateTime ModifiedTime { get; set; }
    public bool IsDirectory { get; set; }
    public bool IsParentDirectory { get; set; }  // Special flag for ".." entry

    public string FormattedSize => IsParentDirectory ? "" : (IsDirectory ? "<DIR>" : FormatSize(Size));

    private static string FormatSize(long bytes)
    {
        string[] sizes = { "B", "KB", "MB", "GB" };
        int order = 0;
        double size = bytes;
        while (size >= 1024 && order < sizes.Length - 1)
        {
            order++;
            size /= 1024;
        }
        return $"{size:0.##} {sizes[order]}";
    }
}

File: android-client/app/src/main/java/com/bluelink/transfer/FileItem.kt
package com.bluelink.transfer

data class FileItem(
    val name: String,
    val size: Long,
    val modifiedTime: String,
    val isDirectory: Boolean,
    val isParentDirectory: Boolean = false
) {
    val formattedSize: String
        get() = if (isDirectory) "<DIR>" else formatSize(size)

    private fun formatSize(bytes: Long): String {
        val sizes = listOf("B", "KB", "MB", "GB")
        var size = bytes.toDouble()
        var order = 0
        while (size >= 1024 && order < sizes.size - 1) {
            order++
            size /= 1024
        }
        return "%.2f %s".format(size, sizes[order])
    }
}
